Tools/CISupport/ews-build/steps.py

```diff
@@ -67,6 +72,74 @@
 LAYOUT_TESTS_URL = '{}{}/blob/{}/LayoutTests/'.format(GITHUB_URL, GITHUB_PROJECTS[0], DEFAULT_BRANCH)
 
 
+class TwistedAdditions(object):
+    class Response(object):
+        def __init__(self, status_code, content=None, url=None, headers=None):
+            self.status_code = status_code
+            self.content = content or b''
+            self.url = url
+            self.headers = headers or {}
+
+        @property
+        def text(self):
+            return self.content.decode('utf-8')
+
+        def json(self):
+            return json.loads(self.text)
+
+    class Printer(protocol.Protocol):
+        def __init__(self, finished):
+            self.finished = finished
+            self.data = b''
+
+        def dataReceived(self, bytes):
+            self.data += bytes
+
+        def connectionLost(self, reason):
+            self.finished.callback(self.data)
+
+    @classmethod
+    @defer.inlineCallbacks
+    def request(cls, type=None, url=None, params=None, headers=None, logger=None, timeout=10):
+        logger = logger or (lambda _: None)
+        typ = type or b'GET'
+
+        if not url:
+            logger('No URL provided\n')
+            defer.returnValue(None)
+        if params:
+            url = '{}?{}'.format(url, '&'.join([f'{key}={value}' for key, value in params.items()]))
+
+        headers = headers or {}
+        if 'User-Agent' not in headers:
+            headers['User-Agent'] = ['python-twisted/{}'.format(twisted.__version__)]
+
+        try:
+            agent = Agent(reactor, connectTimeout=timeout)
+            response = yield agent.request(typ, url.encode('utf-8'), Headers(headers))
+            finished = defer.Deferred()
+            response.deliverBody(cls.Printer(finished))
+            data = yield finished
+
+            headers = {}
+            for key, values in response.headers.getAllRawHeaders():
+                headers[key.decode('utf-8')] = [value.decode('utf-8') for value in values]
+
+            defer.returnValue(cls.Response(
+                status_code=response.code,
+                headers=headers,
+                content=data,
+                url=url,
+            ))
+
+        except error.ConnectError as e:
```

Comment:
Since we are construing url based on data contained in params , should we catch exception for the case when url is invalid? 
Similar to https://ews-build.webkit.org/#/builders/70/builds/23670/steps/17/logs/err_text
`builtins.ValueError: Invalid URI b'https://results.webkit.org/api/results-summary/layout-tests/gamepad/gamepad-polling-access.html?platform=mac&style=release&flavor=wk2&ref=256843.3@(HEAD detached at 1d7abcd180ab)'`